android/java/org/chromium/chrome/browser/appmenu/BraveShieldsMenuAdapter.java

```diff
@@ -0,0 +1,743 @@
+    private OnCheckedChangeListener mBraveShieldsHTTPSEverywhereChangeListener;
+    private Switch mBraveShieldsBlocking3rdPartyCookiesSwitch;
+    private OnCheckedChangeListener mBraveShieldsBlocking3rdPartyCookiesChangeListener;
+    private Switch mBraveShieldsBlockingScriptsSwitch;
+    private OnCheckedChangeListener mBraveShieldsBlockingScriptsChangeListener;
+    private Switch mBraveShieldsFingerprintsSwitch;
+    private OnCheckedChangeListener mBraveShieldsFingerprintsChangeListener;
+
+    private boolean mIncognitoTab;
+    public void setIncognitoTab(final boolean incognitoTab) {
+        mIncognitoTab = incognitoTab;
+    }
+
+    public BraveShieldsMenuAdapter(String host,
+            String title,
+            List<MenuItem> menuItems,
+            LayoutInflater inflater,
+            BraveShieldsMenuObserver menuObserver,
+            ListPopupWindow popup,
+            int currentDisplayWidth) {
+        mIncognitoTab = false;
+        mMenuItems = menuItems;
+        mInflater = inflater;
+        mDpToPx = inflater.getContext().getResources().getDisplayMetrics().density;
+        mMenuObserver = menuObserver;
+        mPositionViews = new SparseArray<View>();
+        mPopup = popup;
+        mCurrentDisplayWidth = currentDisplayWidth;
+        mHost = host;
+        mTitle = title;
+    }
+
+    public void updateHost(String host) {
+        mHost = host;
+    }
+
+    @Override
+    public boolean isEnabled(int position) {
+        if (position >= 1 &&  position <= 8) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public int getCount() {
+        return mMenuItems.size();
+    }
+
+    @Override
+    public int getViewTypeCount() {
+        return VIEW_TYPE_COUNT;
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        MenuItem item = getItem(position);
+        int viewCount = item.hasSubMenu() ? item.getSubMenu().size() : 1;
+
+        if (item.getItemId() == R.id.update_menu_id) {
+            return UPDATE_MENU_ITEM;
+        } else if (viewCount == 4) {
+            return FOUR_BUTTON_MENU_ITEM;
+        } else if (viewCount == 3) {
+            return THREE_BUTTON_MENU_ITEM;
+        } else if (viewCount == 2) {
+            return TITLE_BUTTON_MENU_ITEM;
+        }
+        return STANDARD_MENU_ITEM;
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return getItem(position).getItemId();
+    }
+
+    @Override
+    public MenuItem getItem(int position) {
+        if (position == ListView.INVALID_POSITION) return null;
+        assert position >= 0;
+        assert position < mMenuItems.size();
+        return mMenuItems.get(position);
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        final MenuItem item = getItem(position);
+        StandardMenuItemViewHolder holder = null;
+        switch (getItemViewType(position)) {
```

Comment:
this is a little confusing to read. There is a switch statement with just one `case` and a `default`, but then an `if` statement nested inside it with a bunch of different options. It seems like it would be more readable if it was reversed and had a switch inside an if/else? Maybe also move the long if/else/case into another method?

Reply:
Removed the switch